chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -301,16 +301,33 @@
     # make sure the repos are properly setup
     repos = Provisioner::Repositories.get_repos(node[:platform], node[:platform_version], node[:kernel][:machine])
     for name, attrs in repos
-      url = %x{zypper --non-interactive repos #{name} 2> /dev/null | grep "^URI " | cut -d : -f 2-}
-      url.strip!
-      if url != attrs[:url]
-        unless url.empty?
+      current_url = nil
+      current_priority = nil
+
+      out = %x{LANG=C zypper --non-interactive repos #{name} 2> /dev/null}
+      out.split("\n").each do |line|
+        attribute, value = line.split(":", 2)
+        next if value.nil?
+        attribute.strip!
+        value.strip!
+        if attribute == "URI"
```

Comment:
Metrics/BlockNesting: Avoid more than 3 levels of block nesting. (https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)